### PR TITLE
Allow disabling of apiserver endpoints reconciler

### PR DESCRIPTION
**Pull request by thz**
This change introduces .apiserver.endpointReconcilingDisabled in
componentsOverride of the Cluster CR. When enabled the apiserver
is started with --endpoint-reconciler-type=none which disables
the reconciling of the apiserver endpoint (default: false).
This change also introduces the flag --apiserver-endpoint-reconciling-disabled
for kubermatic-controller-manager which lets the
clusterComponentDefaulter controller default .endpointReconcilingDisabled
to true for all Clusters.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Allow disabling of apiserver endpoint reconciling.
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -245,13 +245,18 @@ type AuditLoggingSettings struct {
...
-	Apiserver         DeploymentSettings  `json:"apiserver"`
+	Apiserver         APIServerSettings   `json:"apiserver"`
 	ControllerManager DeploymentSettings  `json:"controllerManager"`
 	Scheduler         DeploymentSettings  `json:"scheduler"`
 	Etcd              StatefulSetSettings `json:"etcd"`
 	Prometheus        StatefulSetSettings `json:"prometheus"`
 }
 
+type APIServerSettings struct {
+	DeploymentSettings
```
Comment (alvaroaleman) [inline api/pkg/crd/kubermatic/v1/cluster.go]: This needs a ``json:",inline"`` tag

**Review by alvaroaleman — COMMENTED** [review]
Please also add the option and a default to the chart template in config/kubermatic/templates/kubermatic-controller-manager-dep.yaml and config/kubtermatic/values.yaml, respectively.

**Comment by toschneck** [comment]
@thz please also describe the use case and the functionality in the documentation. i think it's not so easy for end user to see for what this flag can be used

**Pull request merged**